v2/internal/testcommon/direct_client.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package testcommon
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+type directClient struct {
+	inner client.Client
+}
+
+// NewTestClient is a thin wrapper around controller-runtime client.New, except that we
+// repopulate the objects GVK since for some reason they do that in the cached client and not
+// in the direct one...
```

Comment:
😲 Only in the cached client! 😲

Reply:
Yep - no idea why the behavioral different but this direct client is just a thin wrapper that makes the behave the same.